neurolib/utils/signal.py

```diff
@@ -0,0 +1,515 @@
+"""
+Base classes for representing signals.
+"""
+
+import logging
+from copy import deepcopy
+
+import numpy as np
+import xarray as xr
+from mne import set_log_level
+from mne.filter import filter_data, resample
+from scipy.signal import detrend, get_window, hilbert
+
+NC_EXT = ".nc"
+MNE_LOGGING_LEVEL = "WARNING"
+set_log_level(MNE_LOGGING_LEVEL)
+
+
+class Signal:
+    name = ""
+    label = ""
+    signal_type = ""
+    unit = ""
+    description = ""
+    _copy_attributes = [
+        "name",
+        "label",
+        "signal_type",
+        "unit",
+        "description",
+        "process_steps",
+    ]
+
+    @classmethod
+    def from_file(cls, filename):
```

Comment:
What's the use case scenario of this? Is this so the user can save and load processed signals to disk? Should we have this for model outputs as well?

Reply:
typically you'd do something like

```python
signal = RatesSignal.from_file("filename_of_saved_data.nc")
```

which is shorter and more concise than something like:
```
array = xr.load_dataarray("filename_of_saved_data.nc")
signal = RatesSignal(data=array)
```